pkg/util/bufalloc/byte_allocator.go

```diff
@@ -61,3 +65,26 @@ func (a ByteAllocator) Copy(src []byte, extraCap int) (ByteAllocator, []byte) {
 	copy(alloc, src)
 	return a, alloc
 }
+
+// CopySpan performs a deep copy of the provided Span.
+func (a ByteAllocator) CopySpan(src roachpb.Span) (ByteAllocator, roachpb.Span) {
+	var dst roachpb.Span
+	extraCap := 0
+	if len(src.EndKey) > 0 {
+		a, dst.EndKey = a.Copy(src.EndKey, 0)
+	} else {
+		extraCap = 1
+	}
+	a, dst.Key = a.Copy(src.Key, extraCap)
```

Comment:
Comment that this is a cheaper way to get `key.Next()` and `key` to share the same memory and is kosher because the returned `Span` is immutable. That latter assumption is worth mentioning in the method comment.